client/os.cpp

```diff
@@ -47,9 +47,15 @@ namespace tec {
 
 		// Create a windowed mode window and its OpenGL context.
 		glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
-		glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, glMajor);
-		glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, glMinor);
-		//glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE); // Only with GL 3.2
+    #ifdef __APPLE__
+		glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE); // Apple requires Core for modern GL
+        // Ignore the requested version.  macOS needs 3.3
```

Comment:
Should this be changed to a minimum 3.2 as core begins at 3.2?

A reason it might not work with 3.2 right now is because it does a shader version check at line 90. This is hard-coded to check for shader version 330. What I am asking is if you remove the block from line 90-104 does it compile with core version 3.2?

Reply:
I was working under the recollection that macOS did not implement 3.2.  But I checked the docs again - Apple started at version 3.2 when they added Core support and then added 3.3 support a couple versions later.  So this can be changed.